### ISSUE TITLE
Opening Xdebug tunnel fails

### ISSUE THREAD

**Issue by knowler** [roots/trellis#942]
## Submit a feature request or bug report

- [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/guidelines/blob/master/CONTRIBUTING.md)
- [ ] This is a feature request
- [x] This is a bug report
- [x] This request isn't a duplicate of an [existing issue](https://github.com/roots/trellis/issues)
- [x] I've read the [docs](https://roots.io/trellis/docs) and followed them (if applicable)
- [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/c/trellis) forums

Replace any `X` with your information.

---

**What is the current behavior?**

Installed Trellis today and was developing a site locally, however, I cannot open Xdebug tunnel.

**What is the expected or desired behavior?**

Debugging with Xdebug.

---

## Bug report

**Please provide steps to reproduce, including full log output:**

Running `./bin/xdebug-tunnel.sh open development` or `./bin/xdebug-tunnel.sh open 192.168.50.5`, Ansible fails:

```
PLAY [Test Connection and Determine Remote User] **********************************

TASK [connection : Require manual definition of remote-user] *************************
skipping: [192.168.50.5]

TASK [connection : Specify preferred HostKeyAlgorithms for unknown hosts] *********
skipping: [192.168.50.5]

TASK [connection : Check whether Ansible can connect as root] **********************
ok: [192.168.50.5 -> localhost]

TASK [connection : Warn about change in host keys] **********************************
skipping: [192.168.50.5]

TASK [connection : Set remote user for each host] ************************************
ok: [192.168.50.5]

TASK [connection : Announce which user was selected] *******************************
Note: Ansible will attempt connections as user = Knowler
ok: [192.168.50.5]

TASK [connection : Load become password] *******************************************
ok: [192.168.50.5]

PLAY [Enable or Disable Xdebug and SSH Tunnel] **************************************

TASK [Gathering Facts] *****************************************************************
System info:
  Ansible 2.4.2.0; Darwin
  Trellis at "Update `vagrant_box_version` to `>= 201801.02.0`"
---------------------------------------------------
MODULE FAILURE
sudo: a password is required

fatal: [192.168.50.5]: FAILED! => {"changed": false, "module_stdout": "", "rc": 1}
        to retry, use: --limit @/Users/knowler/Sites/united/trellis/xdebug-tunnel.retry

PLAY RECAP ***************************************************************************
192.168.50.5               : ok=4    changed=0    unreachable=0    failed=1
```

After running `vagrant reload` and then running the command again will go a bit further, but still fails:

```
PLAY [Test Connection and Determine Remote User] *********************************************************************************************************************************************************************************************

TASK [connection : Require manual definition of remote-user] *********************************************************************************************************************************************************************************
skipping: [192.168.50.5]

TASK [connection : Specify preferred HostKeyAlgorithms for unknown hosts] ********************************************************************************************************************************************************************
skipping: [192.168.50.5]

TASK [connection : Check whether Ansible can connect as root] ********************************************************************************************************************************************************************************
ok: [192.168.50.5 -> localhost]

TASK [connection : Warn about change in host keys] *******************************************************************************************************************************************************************************************
skipping: [192.168.50.5]

TASK [connection : Set remote user for each host] ********************************************************************************************************************************************************************************************
ok: [192.168.50.5]

TASK [connection : Announce which user was selected] *****************************************************************************************************************************************************************************************
Note: Ansible will attempt connections as user = Knowler
ok: [192.168.50.5]

TASK [connection : Load become password] *****************************************************************************************************************************************************************************************************
ok: [192.168.50.5]

PLAY [Enable or Disable Xdebug and SSH Tunnel] ***********************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************************************************************************
ok: [192.168.50.5]

TASK [xdebug : Install Xdebug] ***************************************************************************************************************************************************************************************************************
System info:
  Ansible 2.4.2.0; Darwin
  Trellis at "Update `vagrant_box_version` to `>= 201801.02.0`"
---------------------------------------------------
[Errno 2] No such file or directory
fatal: [192.168.50.5]: FAILED! => {"changed": false, "cmd": "apt-get update", "rc": 2}
        to retry, use: --limit @/Users/knowler/Sites/united/trellis/xdebug-tunnel.retry

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
192.168.50.5               : ok=5    changed=0    unreachable=0    failed=1
```

I tried this on another fresh Trellis install and encountered the same issue. Let me know if you need anymore details.

**Please describe your local environment:**

Ansible version: 2.4.2.0

OS: macOS 10.13.2

Vagrant version: 2.0.1

**Where did the bug happen? Development or remote servers?**

Development

**Please provide a repository or your `wordpress_sites` config (if possible):**

```yml
# group_vars/development/wordpress_sites.yml
wordpress_sites:
  equipping.ca:
    site_hosts:
      - canonical: equipping.test
        redirects:
          - www.equipping.test
    local_path: ../equipping.ca
    admin_email: admin@equipping.test
    multisite:
      enabled: false
    ssl:
      enabled: false
      provider: self-signed
    cache:
      enabled: false
```

**Is there a related [Discourse](https://discourse.roots.io/) thread or were any utilized (please link them)?**

No.

**Comment by swalkinshaw** [comment]
`[Errno 2] No such file or directory` is a really strange error to have during `apt-get update` 🤔 

Your other Trellis box had the same error? Or another one?

**Comment by fullyint** [comment]
I think the issue could have to do with running `xdebug-tunnel.sh` with local development, whereas I think it is only needed for remotes like staging or production.

The [docs](https://roots.io/trellis/docs/debugging-php/#using-xdebug-in-production) mention:
>While we default to installing Xdebug in development, installing it in any other environment is "opt-in." **It is not recommended to use Xdebug in production**, but ... [if you must] ... This is where `bin/xdebug-tunnel.sh` comes in.

**Comment by knowler** [comment]
@swalkinshaw It was the same set of errors in the other Trellis box. I tried it with a box on Windows and it worked fine.

@fullyint I see the error of my ways. I was thinking I had to enable it, but I see it’s enabled by default for dev in `group_vars/development/php.yml`. With a fresh Trellis install, Xdebug works fine for development.

I should have done a bit more research before opening this issue. Though something does seem to be breaking on macOS (still need someone else to confirm it's not just me), like @fullyint said, there isn’t really a use case for this in development. I'll let one of you decide whether you want to close the issue or investigate further. Thanks for your help.
